thebergamo/react-native-fbsdk-next
# chore: Add typescript type definitions

Hello,

Due to the power of Typescript, it is widely used by the community (including me). So to support all Typescript users, I generated typescript type definitions for all current modules.

Steps taken to generate:
1) Use [babel-plugin-flow-to-typescript](https://www.npmjs.com/package/babel-plugin-flow-to-typescript) plugin to transform flow to typescript
2) Transform all require statements into import statements
3) Fix all module.exports into typescript exports
4) Generate type definitions using tsc
4) Remove the declare keyword from all files but index.d.ts
5) Declare module with the same name as the package inside index.d.ts
6) Edit all import statements from "import Foo from './bar'" to "import('./bar').default"

Steps taken to add into package:
1) Create a directory called "types" at the root of the repo
2) Add all type definitions files inside it
3) Add "types" directory under "files" inside package.json
4) Add property "types" inside package.json with value "types/index.d.ts"

Steps taken to test:
1) Install package using git inside another project
2) Check that types are accessible

Credit for the generation go to the babel-plugin-flow-to-typescript babel plugin and this medium [article](https://www.npmjs.com/package/babel-plugin-flow-to-typescript)

Thank you
